### DISCUSSION TITLE
How many requests does argocd make to git?

### DISCUSSION THREAD

**Discussion by zsaladin** [argoproj/argo-cd#10192] [Q&A]
I'm building gitops with argocd.
We use github enterprise internally.

An administrator of the github enterprise told me that your argocd makes too many requests.
The number of the requests was about 1200 per hour.

I checked a few things.
- `timeout.reconciliation` option. But I didn't specify the option. So its value was 3m by default.
- I didn't turn on auto sync.
- The number of apps is just 5~6.

![스크린샷 2022-08-04 오후 1 56 54](https://user-images.githubusercontent.com/6466389/182766315-2b45177d-f4ba-4625-a511-62e2191c9000.png)
![스크린샷 2022-08-04 오후 1 57 09](https://user-images.githubusercontent.com/6466389/182766342-e49df019-116d-4f03-80c0-0c44c614edce.png)

I think Git requests are normal. But reconciliation activity spikes when I sync an app manually. 

Can you give some advise on reducing the number of requests?
Or is it normal situation?

helm chart - 4.10.0

**Comment by bamakos-finest** [comment]
I think first you should find out from your GitHub admin how many requests per hour the instance can handle? It could also be helpful to consider how many users the instance is meant to support.

Alternatively you could mirror your GitOps repo from your private GitHub instance to a managed Git provider (i.e. AWS CodeCommit) and then point Argo CD to the mirrored repository in order to reduce load on your private GitHub instance.
